pisek/jobs/parts/solution.py

```diff
@@ -229,31 +256,99 @@ def _jobs_points(self) -> list[float]:
+
+        if verbosity == 0:
+            text += f"{predecessor_summary}{verdict_marks(self.new_jobs)}"
+
+        elif verbosity == 1:
+            max_sub_name_len = max(
+                len(subtask_name(num)) for num in self._env.config.subtasks
             )
-        )
-        for verdict in Verdict:
-            count = previous.count(verdict)
-            if count > 0:
-                s += f"{count}{verdict.mark()}"
-        s += ") "
-        if s == "() ":
-            s = ""
-
-        for job, result in zip(self.new_jobs, self._job_results(self.new_jobs)):
-            if job.state == State.canceled:
-                s += "-"
-            elif result is None:
-                s += " "
-            else:
-                s += str(result)
-
-        return s
+            max_sub_points_len = max(
+                len(str(sub.points)) for sub in self._env.config.subtasks.values()
+            )
+
+            subtask_name = pad(self.subtask.name + ":", max_sub_name_len + 1)
+            subtask_points = pad_left(format_points(self._points), max_sub_points_len)
+
+            text += right_aligned_text(
+                f"{subtask_name} {subtask_points}p  "
+                f"{predecessor_summary}{verdict_marks(self.new_jobs)} ",
+                f"slowest {self.slowest_time:.2f}s",
+                offset=-2,
+            )
+
+        elif verbosity >= 2:
+            longest_input_name = max(map(lambda j: len(j.input.name), self.new_jobs))
+            subtask_info = (
+                right_aligned_text(
+                    f"{self.subtask.name}: {format_points(self._points)}p ",
+                    f"slowest {self.slowest_time:.2f}s",
+                    offset=-2,
+                )
+                + "\n"
+            )
+            text += colored_env(subtask_info, "magenta", self._env)
+
+            max_pred_name_len = max(
+                (len(subtask_name(pred)) for pred in self.subtask.predecessors),
+                default=0,
+            )
+            for pred in self.subtask.predecessors:
+                pred_group = all_subtasks[pred]
+                text += right_aligned_text(
+                    tab(
+                        f"Predecessor {pad(subtask_name(pred) + ':', max_pred_name_len + 1)}  "
+                        f"{pred_group.status(all_subtasks, 0)} "
+                    ),
+                    f"slowest {pred_group.slowest_time:.2f}s",
+                    offset=-2,
+                )
+                text += "\n"
+
+            if len(self.subtask.predecessors) and any(
+                map(lambda j: j.result, self.new_jobs)
+            ):
+                text += "\n"
+
+            for job in self.new_jobs:
+                if job.result is not None:
+                    input_verdict = tab(
+                        f"{pad(job.input.name + ':', longest_input_name+1)} "
+                        f"{pad(job.verdict_text(), Verdict.pad_length())} "
```

Comment:
Is there a reason this needs to be padded? `right_aligned_text` will add spaces anyways.